skyportal/handlers/api/user.py

```diff
@@ -359,6 +361,51 @@ def post(self):
         self.verify_and_commit()
         return self.success(data={"id": user_id})
 
+    @permissions(["Manage users"])
+    def patch(self, user_id):
+        """
+        ---
+        description: Update a User record
+        tags:
+          - users
+        parameters:
+          - in: path
+            name: user_id
+            required: true
+            schema:
+              type: integer
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  expirationDate:
+                    type: string
+                    description: |
+                      Arrow-parseable date string (e.g. 2020-01-01). Set a user's expiration
+                      date, after which the user is transitioned into a viewonly role.
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+        """
+        data = self.get_json()
+        if user_id is not None:
+            user = User.get_if_accessible_by(
+                user_id, self.current_user, mode="update", raise_if_none=True
+            )
+            if data.get("expirationDate") is not None:
+                expiration_date = data.pop("expirationDate").strip()
+                expiration_date = arrow.get(expiration_date).datetime
+                user.expiration_date = expiration_date
```

Comment:
```suggestion
            if (expiration_date := data.get("expirationDate")) is not None:
                user.expiration_date = arrow.get(expiration_date.strip()).datetime
```
Just a minor compactification, and an opportunity to add our first walrus operator to SkyPortal! 😁 🎉